mitosheet/src/mito/components/endo/GridData.tsx

```diff
@@ -100,12 +111,46 @@ const GridData = (props: {
                                         ...getBorderStyle(props.gridState.selections, props.gridState.copiedSelections, rowIndex, columnIndex, sheetData.numRows, matchesSearch, props.uiState.highlightedColumnIndex),
                                         ...(conditionalFormat || {})
                                     }}
+                                    onContextMenu={(e) => {
+                                        e.preventDefault();
+                                        e.stopPropagation();
+                                        props.setUIState((prevUiState) => {
+                                            return {
+                                                ...prevUiState,
+                                                currOpenDropdown: {
+                                                    rowIndex: rowIndex,
+                                                    columnIndex: columnIndex
+                                                }
+                                            }
+                                        });
+                                        props.setGridState((prevGridState: GridState) => {
+                                            return {
+                                                ...prevGridState,
+                                                selections: [{
+                                                    startingRowIndex: rowIndex,
+                                                    endingRowIndex: rowIndex,
+                                                    startingColumnIndex: columnIndex,
+                                                    endingColumnIndex: columnIndex,
+                                                    sheetIndex: props.sheetIndex
+                                                }]
+                                            }
+                                        });
+                                    
+                                    }}
                                     tabIndex={-1}
                                     mito-col-index={columnIndex}
                                     mito-row-index={rowIndex}
                                     title={displayCellData}
                                 >
                                     {displayCellData}
+                                    <CellDropdown
+                                        display={displayDropdown}
+                                        rowIndex={rowIndex}
+                                        colIndex={columnIndex}
+                                        setUiState={props.setUIState}
+                                        actions={props.actions}
+                                        closeOpenEditingPopups={props.closeOpenEditingPopups}
+                                    />
```

Comment:
Are there performance impacts of these? It effectively adds `# of rendered cells` to what we render -- which might be a bit on larger screens.

Reply:
I had this thought as well. What if we moved the display state into `GridData.tsx` and then only rendered it if it needs to be rendered?

Reply:
That sounds reasonable to me! I wonder if the dropdown needs the anchor there (honestly, no idea) -- but why don't you give it a try?